PR: https://github.com/odoo/odoo/pull/200949

From: 9fcea4c9385237c8465e5409dd7515f138681c7d
From: sako-odoo
Date: 2026-05-11 06:05:52

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = "ir.attachment"
-    attachments_count = fields.Integer('Attachments Count', compute='_compute_attachments_count')
-    _inherit = 'product.document'
-    attached_on_mrp = fields.Selection(

Total Changes: 113

[REM] mrp, product: removed product documents form BoM

Product documents are already accessible from the product form, making their
presence in the BoM form redundant. Removing this section improves clarity and
avoids duplicated information.

Since documents are no longer created specifically for BoMs, this removes the
MRP-related changes; likewise, the `attached_on_mrp` selection field has also
been removed from the product document form view to provide a cleaner UI
for the user.

task: 4593207

Enterprise PR: odoo/enterprise#86200
Upgrade PR: odoo/upgrade#7357

closes odoo/odoo#200949

Signed-off-by: Steve Van Essche <svs@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mrp/models/__init__.py
+++ b/addons/mrp/models/__init__.py
@@ -1,7 +1,6 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from . import product_document
 from . import res_config_settings
 from . import mrp_bom
 from . import mrp_routing

--- a/addons/mrp/models/ir_attachment.py
+++ b/None
@@ -1,18 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import models
-
-
-class IrAttachment(models.Model):
-    _inherit = "ir.attachment"
-
-    def _post_add_create(self, **kwargs):
-        super()._post_add_create(**kwargs)
-        if self.res_model == "mrp.bom":
-            bom = self.env['mrp.bom'].browse(self.res_id)
-            self.res_model = bom.product_id._name if bom.product_id else bom.product_tmpl_id._name
-            self.res_id = bom.product_id.id if bom.product_id else bom.product_tmpl_id.id
-            self.env['product.document'].create({
-                'ir_attachment_id': self.id,
-                'attached_on_mrp': 'bom'
-            })

--- a/addons/mrp/models/mrp_bom.py
+++ b/addons/mrp/models/mrp_bom.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models, _
 from odoo.exceptions import AccessError, UserError, ValidationError
 from odoo.fields import Command, Domain
 from odoo.tools import float_compare
-from odoo.tools.misc import clean_context, OrderedSet
+from odoo.tools.misc import clean_context
 
 from collections import defaultdict
 
@@ -559,31 +559,6 @@ class MrpBom(models.Model):
 
         return price_unit
 
-    # -------------------------------------------------------------------------
-    # DOCUMENT
-    # -------------------------------------------------------------------------
-
-    def _get_mail_thread_data_attachments(self):
-        res = super()._get_mail_thread_data_attachments()
-        return res | self._get_extra_attachments()
-
-    def _get_extra_attachments(self):
-        is_byproduct = self.env.user.has_group('mrp.group_mrp_byproducts')
-        product_ids, template_ids = OrderedSet(), OrderedSet()
-        for bom in self:
-            product_ids.add(bom.product_id.id)
-            template_ids.add(bom.product_tmpl_id.id)
-            if is_byproduct:
-                product_ids.update(bom.byproduct_ids.product_id.ids)
-                template_ids.update(bom.byproduct_ids.product_id.product_tmpl_id.ids)
-
-        domain = Domain('attached_on_mrp', '=', 'bom') & (
-            (Domain('res_model', '=', 'product.product') & Domain('res_id', 'in', product_ids))
-            | (Domain('res_model', '=', 'product.template') & Domain('res_id', 'in', template_ids))
-        )
-        attachements = self.env['product.document'].search(domain).ir_attachment_id
-        return attachements
-
     @api.model
     def _skip_for_no_variant(self, product, bom_attribule_values, never_attribute_values=False):
         """ Controls if a Component/Operation/Byproduct line should be skipped based on the 'no_variant' attributes
@@ -717,7 +692,6 @@ class MrpBomLine(models.Model):
     child_line_ids = fields.One2many(
         'mrp.bom.line', string="BOM lines of the referred bom",
         compute='_compute_child_line_ids')
-    attachments_count = fields.Integer('Attachments Count', compute='_compute_attachments_count')
     tracking = fields.Selection(related='product_id.tracking')
     bom_code = fields.Char(related='bom_id.code')
     bom_type = fields.Selection(related='bom_id.type')
@@ -738,16 +712,6 @@ class MrpBomLine(models.Model):
             else:
                 line.child_bom_id = bom_by_product.get(line.product_id, False)
 
-    @api.depends('product_id')
-    def _compute_attachments_count(self):
-        for line in self:
-            nbr_attach = self.env['product.document'].search_count([
-                '&', '&', ('attached_on_mrp', '=', 'bom'), ('active', '=', True),
-                '|',
-                '&', ('res_model', '=', 'product.product'), ('res_id', '=', line.product_id.id),
-                '&', ('res_model', '=', 'product.template'), ('res_id', '=', line.product_tmpl_id.id)])
-            line.attachments_count = nbr_attach
-
     @api.depends('child_bom_id')
     def _compute_child_line_ids(self):
         """ If the BOM line refers to a BOM, return the ids of the child BOM lines """
@@ -793,39 +757,6 @@ class MrpBomLine(models.Model):
 
         return self.env['mrp.bom']._skip_for_no_variant(product, self.bom_product_template_attribute_value_ids, never_attribute_values)
 
-    def action_see_attachments(self):
-        domain = [
-            '&', ('attached_on_mrp', '=', 'bom'),
-            '|',
-            '&', ('res_model', '=', 'product.product'), ('res_id', '=', self.product_id.id),
-            '&', ('res_model', '=', 'product.template'), ('res_id', '=', self.product_id.product_tmpl_id.id)]
-        attachments = self.env['product.document'].search(domain)
-        nbr_product_attach = len(attachments.filtered(lambda a: a.res_model == 'product.product'))
-        nbr_template_attach = len(attachments.filtered(lambda a: a.res_model == 'product.template'))
-        context = {'default_res_model': 'product.product',
-            'default_res_id': self.product_id.id,
-            'default_company_id': self.company_id.id,
-            'attached_on_bom': True,
-            'search_default_context_variant': not (nbr_product_attach == 0 and nbr_template_attach > 0) if self.env.user.has_group('product.group_product_variant') else False
-        }
-
-        return {
-            'name': _('Attachments'),
-            'domain': domain,
-            'res_model': 'product.document',
-            'type': 'ir.actions.act_window',
-            'view_mode': 'kanban,list,form',
-            'target': 'current',
-            'help': _('''<p class="o_view_nocontent_smiling_face">
-                        Upload files to your product
-                    </p><p>
-                        Use this feature to store any files, like drawings or specifications.
-                    </p>'''),
-            'limit': 80,
-            'context': context,
-            'search_view_id': self.env.ref('product.product_document_search').ids
-        }
-
     def action_open_parent_bom(self):
         self.ensure_one()
         return {

--- a/addons/mrp/models/product_document.py
+++ b/None
@@ -1,23 +0,0 @@
-
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import fields, models
-
-
-class ProductDocument(models.Model):
-    _inherit = 'product.document'
-
-    def _default_attached_on_mrp(self):
-        return "bom" if self.env.context.get('attached_on_bom') else "hidden"
-
-    attached_on_mrp = fields.Selection(
-        selection=[
-            ('hidden', "Hidden"),
-            ('bom', "Bill of Materials")
-        ],
-        required=True,
-        string="MRP : Visible at",
-        help="Leave hidden if document only accessible on product form.\n"
-            "Select Bill of Materials to visualise this document as a product attachment when this product is in a bill of material.",
-        default=lambda self: self._default_attached_on_mrp(),
-    )
